quic/s2n-quic-platform/src/io/tokio/builder.rs

```diff
@@ -175,6 +195,14 @@ impl Builder {
     }
 
     pub fn build(self) -> io::Result<Io> {
+        ensure!(
+            self.mtu_config.is_valid(),
+            Err(io::Error::new(
+                io::ErrorKind::InvalidInput,
+                "the MTU configuration must have min_mtu <= initial_mtu <= max_mtu"
+            ))
+        );
```

Comment:
I wonder if we need a mtu config builder with `Option`s for each field? This would make it so if, for example, you set the `max_mtu` to a value lower than the default `initial_mtu` it wouldn't error. Otherwise, you kind of have to know all of the default values.

Reply:
Currently its not possible to have a `max_mtu` that is lower than the default `initial_mtu`, because the default `initial_mtu` is the `MINIMUM_MTU`, and the `MaxMtu` type cannot be created with a value lower than that. 

Of course this would not hold true if we ever change the default `initial_mtu`, and even prior to that, you could have someone try to configure `base_mtu` without configuring `initial_mtu`, which would always fail :-/ 

All this to say... yeah I guess we do need a builder :-)

Reply:
I added the builder, now if the user configures a `BaseMtu` higher than the default without configuring the `InitialMtu`, the `InitialMtu` will automatically be adjusted to the `BaseMtu`. 

I decided to leave most other cases as `MtuError`s , such as the user configured conflicting MTU settings (such as configuring baseMTU = 1400 and initialMTU = 1300) and also the case where the user configured the BaseMTU or InitialMTU to be higher than the default MaxMTU. My reasoning was that for these cases, its better that the user be aware they made a bad configuration rather than trying to automatically correct their configuration. I added the default values for all the errors the `MtuError` so now it shows as:

`MTU must have 1228 <= base_mtu (default: 1228) <= initial_mtu (default: 1228) <= max_mtu (default: 1500)`